sa/proto/sa.proto

```diff
@@ -0,0 +1,151 @@
+syntax = "proto2";
+
+package sa;
+option go_package = "proto";
+
+import "core/proto/core.proto";
+
+service StorageAuthority {
+        // Getters
+        rpc GetRegistration(RegistrationID) returns (core.Registration) {}
+        rpc GetRegistrationByKey(JsonWebKey) returns (core.Registration) {}
+        rpc GetAuthorization(AuthorizationID) returns (core.Authorization) {}
+        rpc GetValidAuthorizations(GetValidAuthorizationsRequest) returns (ValidAuthorizations) {}
+        rpc GetCertificate(Serial) returns (core.Certificate) {}
+        rpc GetCertificateStatus(Serial) returns (CertificateStatus) {}
+        rpc CountCertificatesRange(Range) returns (Count) {}
+        rpc CountCertificatesByNames(CountCertificatesByNamesRequest) returns (CountByNames) {}
+        rpc CountRegistrationsByIP(CountRegistrationsByIPRequest) returns (Count) {}
+        rpc CountPendingAuthorizations(RegistrationID) returns (Count) {}
+        rpc GetSCTReceipt(GetSCTReceiptRequest) returns (SignedCertificateTimestamp) {}
+        rpc CountFQDNSets(CountFQDNSetsRequest) returns (Count) {}
+        rpc FQDNSetExists(FQDNSetExistsRequest) returns (Exists) {}
+        // Adders
+        rpc NewRegistration(core.Registration) returns (core.Registration) {}
+        rpc UpdateRegistration(core.Registration) returns (core.Empty) {}
+        rpc NewPendingAuthorization(core.Authorization) returns (core.Authorization) {}
+        rpc UpdatePendingAuthorization(core.Authorization) returns (core.Empty) {}
+        rpc FinalizeAuthorization(core.Authorization) returns (core.Empty) {}
+        rpc MarkCertificateRevoked(MarkCertificateRevokedRequest) returns (core.Empty) {}
+        rpc AddCertificate(core.CertificateRequest) returns (AddCertificateResponse) {}
+        rpc AddSCTReceipt(SignedCertificateTimestamp) returns (core.Empty) {}
+        rpc RevokeAuthorizationsByDomain(RevokeAuthorizationsByDomainRequest) returns (RevokeAuthorizationsByDomainResponse) {}
+        rpc DeactivateRegistration(RegistrationID) returns (core.Empty) {}
+        rpc DeactivateAuthorization(AuthorizationID) returns (core.Empty) {}
+}
+
+message RegistrationID {
+        optional int64 id = 1;
+}
+
+message JsonWebKey {
+        optional bytes jwk = 1; 
+}
+
+message AuthorizationID {
+        optional string id = 1;
+}
+
+message GetValidAuthorizationsRequest {
+        optional int64 registrationID = 1;
+        repeated string domains = 2;
+        optional int64 now = 3; // Unix timestamp (nanoseconds)
+}
+
+message ValidAuthorizations {
+        message MapElement {
+                optional string domain = 1;
+                optional core.Authorization authz = 2;
+        }
+        repeated MapElement valid = 1;
```

Comment:
I think all we would need to do is update the boulder-tools image, as far as I know it should be fully backward compatible with what we have now. I think the only really annoying thing is we would need to change from installing the compiler from an apt repo to cloning it from github and building it from source.

Most of what is introduced in 3.0 is new language support and `proto3`, although there are also some bug fixes. While I think we should upgrade eventually I don't see any major benefit in doing so immediately.